go/vt/sqlparser/sql.y

```diff
@@ -717,9 +720,81 @@ base_select:
   {
     $$ = $1
   }
+| set_op
+  {
+    $$ = $1
+  }
+
+set_op:
```

Comment:
This is clever but might be confusing for someone without context. It might be helpful to give an example like `{1,2} ∩  {2,3} U {3,4}` has two results depending on whether you execute the intersect or union first (same as addition/multiplication). Here is 3 sentences that make it clear how we enforce INTERSECT being executed before UNION/EXCEPT...

Reply:
Agreed. It's clever but I'm still not entire sure I understand how it works. Some documentation would be really helpful.